libass/ass_shaper.c

```diff
@@ -956,6 +956,16 @@ int ass_shaper_shape(ASS_Shaper *shaper, TextInfo *text_info)
         ass_shaper_skip_characters(text_info);
 #endif
 
+    // Make sure we don't lose any run breaks due to skipped glyphs
+    for (i = 0; i < text_info->length; i++) {
```

Comment:
It should be possible to rearrange checks without nested loops. Something like
```C
bool new_run = true;
for (int i = 0; i < text_info->length; i++) {
    if (glyphs[i].starts_new_run)
        new_run = true;
    if (glyphs[i].skip)
        continue;
    glyphs[i].starts_new_run = new_run;
    new_run = false;
}
```

Reply:
Is that better though? It seems harder to understand.

Reply:
For me it's instead simpler to understand than nested loops with a common index variable, but feel free to ignore me.
I think that triple copy-paste of condition `i < text_info->length` isn't that great.